tests/foreman/ui/test_contenthost.py

```diff
@@ -129,6 +129,16 @@ def set_ignore_facts_for_os(value=False):
     ignore_setting.update({'value'})
 
 
+@pytest.fixture(scope='module')
+def set_remote_execution_by_default(value=True):
+    """Fixture to set 'remote_execution_by_default' setting"""
+    remote_execution_setting = entities.Setting().search(
+        query={'search': 'name="remote_execution_by_default"'}
+    )[0]
+    remote_execution_setting.value = str(value)
+    remote_execution_setting.update({'value'})
```

Comment:
There's actually already a fixture that does this, you just have to indirectly parametrize `remote_execution_by_default`. You then use the returned object to modify the setting to the value you want.

Reply:
Looks like that is doing it for every test, not at a module level.

Reply:
@swadeley in that case, it might be best to separate the functionality from the function-level fixture and use it for both function and module-level fixtures. We can talk specifics if you'd like

Reply:
Hi, I made this new version, based on @mshriver comment,  before I saw your new comment @JacobCallahan : https://github.com/SatelliteQE/robottelo/pull/9465

Reply:
On second thoughts, that method is no good because it would leave Satellite in a non-REX-by-default state, which we do not want if its a new install. Its only in the upgrade case that that would be acceptable.

Reply:
> should this fixture get the original value and set it back after a yield?

that would not be good if the test ran on a new install.

Reply:
@swadeley wouldn't it then just reset it to the previous (default) value?

Reply:
@JacobCallahan "separate the functionality from the function-level fixture" my fixture, in this PR, or the one[1] Mike linked too?

[1] https://github.com/SatelliteQE/robottelo/blob/master/pytest_fixtures/component/settings.py#L7